### PR TITLE
Landing Page with Preview API Demonstration

### PR DESCRIPTION
**Pull request by jhancock532**
This branch tests some core API functionality; pulling data from Contentful, and enabling Contentful's preview mode by clicking "Open Preview" in the Contentful editor.

### CODE REVIEW

**Review by Morsey187 — CHANGES_REQUESTED** [review]

File: lib/api.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+//To be refactored when design is confirmed.
+//See https://github.com/vercel/next.js/blob/41f87abdf7be4519e7d928bbed4dec314fcd7851/examples/cms-contentful/lib/api.js#L46
+
+async function fetchGraphQL(query = "", preview = false) {
+    return fetch(
+        `https://graphql.contentful.com/content/v1/spaces/${process.env.CONTENTFUL_SPACE_ID}`,
```
[the remark this answers is not in the document]
Comment (jhancock532) [reply lib/api.ts:6]: Good idea, set this up :)

File: lib/api.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+//To be refactored when design is confirmed.
+//See https://github.com/vercel/next.js/blob/41f87abdf7be4519e7d928bbed4dec314fcd7851/examples/cms-contentful/lib/api.js#L46
+
+async function fetchGraphQL(query = "", preview = false) {
```
[the remark this answers is not in the document]
Comment (jhancock532) [reply lib/api.ts:4]: I'm currently using Contentful's Graphql explorer, but we could definitely add a ticket for this

**Comment by jhancock532** [comment]
I've added an early draft of the RichText renderer to this branch as it relies on data from Contentful to render elements. 

I'm not sure about my implementation as there wasn't any documentation about how to make this using TypeScript - be especially careful of the weird type requirements I encountered (I've commented these where appropriate)

I've decided on three color themes for this component to render, based on the existing figma designs. I've used styled components for this, it looks a bit awkward at the moment - there's potential for some refactoring here.

File: components/RichText.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                        value: "We’re always excited about finding new talent and meeting people that are as eager as we are to drive significant positive change.",
+                        nodeType: "text",
+                    },
+                ],
+                nodeType: "paragraph",
+            },
+        ],
+        nodeType: "document",
+    },
+};
```
Comment (jhancock532) [inline components/RichText.tsx:114]: Unnecessary testing data, to be deleted (or converted into a Jest test)

File: components/RichText.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                    {children}
+                </a>
+            ),
+        },
+    };
+};
+
+export const RichText = ({ theme, content }: RichTextProps) => {
+    return (
+        <div className={styles.container}>
```
Comment (jhancock532) [inline components/RichText.tsx:87]: styles.container is unused, we might add style here when we integrate this component into the landing page

File: components/RichText.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+const renderLink = (theme: Theme, children: React.ReactNode) => {
+    switch (theme) {
+        case "LIGHT":
+            return <p className={styles.paragraphLight}>{children}</p>;
+        case "DARK":
+            return <p className={styles.paragraphLight}>{children}</p>;
+        case "INDIGO":
+            return <p className={styles.paragraphLight}>{children}</p>;
+    }
+};
```
Comment (jhancock532) [inline components/RichText.tsx:33]: To be deleted or refactored (I avoided this solution initially as I would have to pass the `node` with an unknown type into the function - however, we could use this function and just pass the node with `: any` instead

File: components/RichText.tsx [hunk 6]
```diff
@@ -0,0 +1,114 @@
+import { documentToReactComponents } from "@contentful/rich-text-react-renderer";
+import { BLOCKS, MARKS, INLINES } from "@contentful/rich-text-types";
+import React from "react";
+import styles from "styles/RichText.module.scss";
+
+type Theme = "LIGHT" | "DARK" | "INDIGO";
```
Comment (jhancock532) [inline components/RichText.tsx:6]: Light for the job post page, dark for the homepage, indigo for the homepage. This should be worth writing documentation about?

File: pages/index.tsx [hunk 7]
```diff
@@ -35,6 +40,7 @@ export default LandingPage;
 
 export async function getStaticProps({ preview = false }) {
     const landingPageContent = (await getLandingPage(preview)) ?? [];
+    console.log("Test 1 ", JSON.stringify(landingPageContent.ctaDescription));
```
Comment (jhancock532) [inline pages/index.tsx:43]: Used for testing, to be deleted.

File: styles/RichText.module.scss [hunk 8]
```diff
@@ -0,0 +1,83 @@
+.bold {
+    font-weight: $weight--bold;
+}
+
+.italic {
+    font-style: italic;
+}
+
+.paragraphLight {
+    color: var(--color--grey);
```
Comment (jhancock532) [inline styles/RichText.module.scss:10]: Line-heights aren't responsive here - refactor the SCSS to include line heights that vary as the font size changes? @include font-size(s) should also include line height?

File: styles/RichText.module.scss [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    color: var(--color--dark-indigo);
+    @include font-size(s);
+    line-height: 30px;
+}
+
+.paragraphDark {
+    color: var(--color--white);
+    @include font-size(s);
+    line-height: 30px;
+}
```
Comment (jhancock532) [inline styles/RichText.module.scss:25]: Refactor as `.paragraphIndigo, .paragraphDark, ... { ... }` for common rules. Or use a seperate `.paragraph` class instead & double up with `styles.paragraph` & `styles.paragraphIndigo`. Or use a more generic utility class based solution with scss?

File: styles/RichText.module.scss [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+.h2 {
+    line-height: 36px;
+    margin-bottom: 30px;
+}
+
+.h3 {
+    line-height: 34px;
+    margin-bottom: 12px;
+}
```
Comment (jhancock532) [inline styles/RichText.module.scss:35]: Taken from designs, ideally refactored into `_variables.scss`

**Review by Morsey187 — APPROVED** [review]
Happy with this, perhaps we can seperate the graphql queires into their own strings to export and import them else where at a later point if that become nescassary.

**Pull request merged**